### PR TITLE
Allow private notes

### LINKED ISSUE #271: Mark issues or notes as private
If you find this interesting

https://github.com/albfan/redmine-java-api/commit/b2951b752273886c4170aa8156d4ba1bf29c8761

I can provide an isolate PR

### PR DESCRIPTION
**Pull request by albfan**
closes #271

### CODE REVIEW

File: src/main/java/com/taskadapter/redmineapi/bean/Issue.java [hunk 1]
```diff
@@ -317,6 +318,17 @@ public void setNotes(String notes) {
         storage.set(NOTES, notes);
     }
 
+    public boolean getPrivateNotes() {
```
Comment (alexeyOnGitHub) [inline src/main/java/com/taskadapter/redmineapi/bean/Issue.java]: the method name should start with "is"

File: src/main/java/com/taskadapter/redmineapi/internal/RedmineJSONBuilder.java [hunk 2]
```diff
@@ -209,21 +209,22 @@ public static void writeIssue(final JSONWriter writer, Issue issue) throws JSONE
 		addIfSet(writer, "spent_hours", storage, Issue.SPENT_HOURS);
 		addIfSet(writer, "assigned_to_id", storage, Issue.ASSIGNEE_ID);
 		addIfSet(writer, "priority_id", storage, Issue.PRIORITY_ID);
-                addIfSet(writer, "done_ratio", storage, Issue.DONE_RATIO);
+		addIfSet(writer, "done_ratio", storage, Issue.DONE_RATIO);
 		addIfSet(writer, "is_private", storage, Issue.PRIVATE_ISSUE);
```
Comment (alexeyOnGitHub) [inline src/main/java/com/taskadapter/redmineapi/internal/RedmineJSONBuilder.java]: formatting-only changes should be in separate commit(s)

File: src/main/java/com/taskadapter/redmineapi/internal/RedmineJSONBuilder.java [hunk 3]
```diff
@@ -209,21 +209,22 @@ public static void writeIssue(final JSONWriter writer, Issue issue) throws JSONE
...
 		addIfSet(writer, "project_id", storage, Issue.PROJECT_ID);
 		addIfSet(writer, "author_id", storage, Issue.AUTHOR_ID);
 		addIfSet(writer, "start_date", storage, Issue.START_DATE, RedmineDateParser.SHORT_DATE_FORMAT_V2.get());
 		addIfSet(writer, "due_date", storage, Issue.DUE_DATE, RedmineDateParser.SHORT_DATE_FORMAT_V2.get());
-                addIfSetIdentifiable(writer, "tracker_id", storage, Issue.TRACKER);
+		addIfSetIdentifiable(writer, "tracker_id", storage, Issue.TRACKER);
```
Comment (alexeyOnGitHub) [inline src/main/java/com/taskadapter/redmineapi/internal/RedmineJSONBuilder.java]: formatting-only changes should be in separate commit(s)

**Comment by albfan** [comment]
I have added javadoc and remove all changes unrelated (formatting)

I need to add some test and as I say, there's no json output flagging a note as private so it is not read when retrieving issues.

**Comment by alexeyOnGitHub** [comment]
great, thank you @albfan

**Comment by albfan** [comment]
Miss the test code. Sorry for that. Maybe would be great to add a code coverage integration (code climate or similar) to refuse PR that decrease code coverage, i.e. don't provide test for new code
